services/QuillLMS/client/app/bundles/Evidence/components/studentView/promptStep.tsx

```diff
@@ -6,7 +6,7 @@ import Feedback from './feedback'
 
 import EditCaretPositioning from '../../helpers/EditCaretPositioning'
 import ButtonLoadingSpinner from '../shared/buttonLoadingSpinner'
-import { highlightSpellingGrammar } from '../../libs/stringFormatting'
+import { highlightGrammar, highlightSpelling, stripEvidenceHtml } from '../../libs/stringFormatting'
```

Comment:
Any reason not to refactor `highlightSpellingGrammar` to handle offsets for all types instead of splitting behavior like this?  I haven't fully thought this question through, so it's possible that there's an obvious reason not to, but my first thought was "we should probably respect offsets on either type".

Reply:
We can't currently do this because the spelling payload does not include an offset (payload schemas differ between grammar and spelling): https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/engines/evidence/lib/evidence/evidence/spelling_check.rb#L69

Reply:
Fair point.  We could just assume that not having `highlight.character` just means an offset of `0` (start at the beginning of the string) or something.  That solves the problem for spelling and future-proofs it if we do decide to add support for it later.  But maybe that's more work than we want to do in this PR?  Up to you.

Reply:
Yeah, let's treat that as out of scope for this PR.